lib/ecl/ecl_file.c

```diff
@@ -992,3 +999,42 @@ static ecl_file_type * ecl_file_iopen_rstblock__( const char * filename , int se
 ecl_file_type * ecl_file_iopen_rstblock( const char * filename , int seqnum_index , int flags) {
   return ecl_file_iopen_rstblock__(filename , seqnum_index , flags );
 }
+
+bool ecl_file_index_valid(const char * file_name, const char * index_file_name) {
+  if ( !(   util_file_exists( file_name ) && util_file_exists (index_file_name)   )   )
+    return false;
+  if (util_file_difftime( file_name , index_file_name) > 0)
+    return false;
+  return true;
+}
+
+void  ecl_file_write_index( ecl_file_type * ecl_file , const char * filename , const char * index_filename) {
```

Comment:
Not necessary to take in the `filename` argument - the `ecl_file` instance knows it's own filename.